### ISSUE TITLE
[bug] light theme font color works poorly with black wallpaper

### ISSUE THREAD

**Issue by wassupluke** [jrpie/Launcher#75]
# Describe the bug

Changing the Color Theme to Light uses the new black font from 68fa4190f378a1f398b4960875e4aa66459a3677, which makes it impossible to see text if wallpaper is black.

# To Reproduce

1. Set wallpaper to a plain black image
2. Go to Launcher app settings
3. Navigate to 'Launcher' tab
4. Click 'Color theme>Light's
5. #000 color text is now impossible to see

# Expected behavior

# Screenshots / Screen recording
https://github.com/user-attachments/assets/10315ccd-4a35-4187-a55a-81fc14301742

# Smartphone (please complete the following information)
 - Device: Pixel 8 Pro
 - Android Version: 15
 - µLauncher Version: j-0.0.17

# Additional info

Incidentally found another bug where the text on the Apps tab disappeared for only the first four entries (see beginning of the screen recording, was unable to reproduce that particular issue or I'd open a separate bug for it)

**Comment by jrpie** [comment]
Thanks for reporting! I'm not too happy with the light theme myself. The main reason to add it, is that the white clock is hard to read on a very light background.
But the proper solution for that is of course to let users select the clock color independently of the theme (see #12)

Regarding the light theme itself, solutions would be:
(a) scrap it (I'm in favor of that)
(b) only allow it together with a solid white ba
(c) 

The problem of the font being unreadable also occurs when combining a solid white wallpaper with the dark or default theme, setting background to transparent and turning text shdow off, so some kind of mitigation might be needed for that as well.

**Comment by wassupluke** [comment]
I don't know how Material You does it but it works well for that, is there
a way to incorporate that without excess headache on your part?

**Comment by jrpie** [comment]
Material you is planned (basically I just need to add a color theme that uses colors from material you, see #31) 
The main problem is that that only works on Android 12+

I don't really want to implement any logic for extracting colors from the wallpaper myself.

**Comment by wassupluke** [comment]
I haven't looked into the color setting logic on this one but I used this for a bit on my laptop a couple years ago and it worked nicely IIRC. https://github.com/dylanaraps/pywal

**Comment by wassupluke** [comment]
I'm just starting to learn basics of the android app development process and am curious if migrating the codebase to jetpack compose would give you the material theming most easily. I realize I'm not the one maintaining this project but am starting to look into the differences between kotlin and compose and potential ease or difficulty of migrating. If jetpack compose is easy enough for you to maintain it would probably be best to make the switch earlier rather than later so there's less to rewrite in the future. Thoughts?

**Comment by jrpie** [comment]
> I haven't looked into the color setting logic on this one but I used this for a bit on my laptop a couple years ago and it worked nicely IIRC. https://github.com/dylanaraps/pywal

Looks like they are using sevaral different libraries / programs to do that, there doesn't seem to be a standard approach: https://github.com/dylanaraps/pywal/tree/master/pywal/backends

But even more problematic than color extraction is the fact that one can not programmaticaly create android themes https://stackoverflow.com/a/63928975 i.e. in order to set the color programmatically, I would need to set the color for every single element or at least create wrappers.

I think I'll do the following:
* scrap the light theme
* let users choose from dark / default theme and additionally a material you theme on android 12+
* let users select a custom color for the clock

**Comment by jrpie** [comment]
> I'm just starting to learn basics of the android app development process and am curious if migrating the codebase to jetpack compose would give you the material theming most easily. I realize I'm not the one maintaining this project but am starting to look into the differences between kotlin and compose and potential ease or difficulty of migrating.
> 
> https://developer.android.com/develop/ui/compose/setup#kotlin
> 
> If jetpack compose is easy enough for you to maintain it would probably be best to make the switch earlier rather than later so there's less to rewrite in the future. Thoughts?

Thanks for the offer! I have not looked into compose yet. On first glance the declarative style looks quite nice, but migrating will be a major effort; I'm not sure whether that is worth the effort (you would need to rewrite basically the entire UI; starting from scratch might be less work).

Knowing how Google treats android devs, I worry about the future of compose. When I started this fork, google had just scrapped some bindings library (which was a fancy new thing when the original app was written). Migrating was painful, even though that was a relatively minor feature. So I'm somewhat concerned about moving to any fancy new thing, especially something as radically different as compose. Are they planning to replace xml by compose entirely? Or is compose considered an alternative style that will be maintained in parallel to xml? (and possibly scrapped as soon as there is another fancy new thing)

Implementing material 3 is definitely possible without migrating to compose (basically the parent theme needs to be changed to `Theme.Material3.Light.NoActionBar` [here](https://github.com/jrpie/Launcher/blob/08f0026dc3de885a331c9c7f0c377b67e4dadb6b/app/src/main/res/values/styles.xml#L4). This seems to break the TabLayout and the PreferenceFragmentCompat also doesn't seem to follow m3, but other than that most things seem to work. So if you are mostly interested in m3, you could try to fix those problems; that would be at least an order of magnitude less work.

For learning new things it is probably best (and more fun), to start with some small project that can be done over the course of a weekend. Starting a migration without very good knowledge of the target is perhaps not a good idea.

That being said, you can of course start to migrate this app and open a PR; it might quite some time for me to learn compose and review the PR though ^^

**Comment by wassupluke** [comment]
Yeah it will take me a while to learn both Kotlin and Compose before I can actually start to be helpful at all lol. The backstory on Google scrapping things is good to know and annoying on the development side of things.

But yeah, I've written zero Kotlin or Compose and this is the first android app I've really gotten into looking at the code and it's making me really curious to figure out how it works.

**Comment by wassupluke** [comment]
It sounds like Jetpack Compose lets you ditch XML and goes pure Kotlin with much less lines of code. 

https://youtu.be/zGP7-VvjmTQ

**Comment by jrpie** [comment]
Yes, it is an entirely different approach to UI. I don't think XML is bad though (I guess I'm just used to it)

**Comment by jrpie** [comment]
A user pointed out on discord, that the light theme is important for E-Ink devices.
Perhaps it could be renamed to E-Ink and include other E-Ink optimizations?
